tsconfig.json

```diff
@@ -1,17 +1,32 @@
 {
   "compilerOptions": {
-    "module": "commonjs",
+    "allowSyntheticDefaultImports": true,
     "esModuleInterop": true,
-    "target": "es6",
-    "noImplicitAny": true,
+    "target": "es2015",
+    "downlevelIteration": true,
+    "importHelpers": true,
+    "sourceMap": true,
+    "module": "commonjs",
     "moduleResolution": "node",
+    "allowJs": false,
+    "checkJs": false,
     "resolveJsonModule": true,
-    "sourceMap": true,
-    "outDir": "dist",
-    "baseUrl": ".",
-    "paths": {
-      "*": ["node_modules/*"]
-    }
+    "lib": ["dom", "es2015", "es2016", "es2017", "esnext"],
+    "forceConsistentCasingInFileNames": true,
+    "experimentalDecorators": true,
+    "noImplicitAny": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noImplicitThis": true,
+    "noUnusedLocals": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "suppressImplicitAnyIndexErrors": true,
+    "types": ["node"],
+    "baseUrl": "./src/",
+    "outDir": "./build",
+    "plugins": [{ "transform": "typescript-transform-paths" }]
   },
-  "include": ["src/**/*"]
+  "include": ["src/**/*.ts", "test/**/*.ts"],
```

Comment:
I think you should include only the src. You can provide just an entry point `main.ts` or `index.ts` and all the TS will be pulled from there.

If you include the tests, it will create aditional folders in the `build` dir when is not required.